src/test/kotlin/no/nav/syfo/oppfolgingstilfelle/kafka/KafkaOppfolgingstilfellePersonServiceSpek.kt

```diff
@@ -643,6 +550,336 @@ class KafkaOppfolgingstilfellePersonServiceSpek : Spek({
                     aktivitetskravList.shouldBeEmpty()
                 }
             }
+            describe("Aktivitetskrav(NY) exists for earlier oppfølgingstilfelle") {
+                val nyAktivitetskrav = Aktivitetskrav.ny(
+                    personIdent = UserConstants.ARBEIDSTAKER_PERSONIDENT,
+                    tilfelleStart = yearAgo,
+                )
+
+                it("updates aktivitetskrav for earlier oppfølgingstilfelle to AUTOMATISK_OPPFYLT and creates Aktivitetskrav(NY) when latest oppfølgingstilfelle lasting 9 weeks (not gradert)") {
+                    database.createAktivitetskrav(nyAktivitetskrav)
+
+                    mockKafkaConsumerOppfolgingstilfellePerson(
+                        kafkaOppfolgingstilfelleNineWeeksNotGradert
+                    )
+
+                    kafkaOppfolgingstilfellePersonService.pollAndProcessRecords(
+                        kafkaConsumer = mockKafkaConsumerOppfolgingstilfellePerson,
+                    )
+
+                    verify(exactly = 1) {
+                        mockKafkaConsumerOppfolgingstilfellePerson.commitSync()
+                    }
+
+                    val aktivitetskravList = database.getAktivitetskrav(
+                        personIdent = UserConstants.ARBEIDSTAKER_PERSONIDENT
+                    )
+
+                    aktivitetskravList.size shouldBeEqualTo 2
+
+                    val (aktivitetskravLatestOppfolgingstilfelle, aktivitetskravEarlierOppfolgingstilfelle) = aktivitetskravList
+                    aktivitetskravEarlierOppfolgingstilfelle.status shouldBeEqualTo AktivitetskravStatus.AUTOMATISK_OPPFYLT.name
+                    aktivitetskravEarlierOppfolgingstilfelle.uuid shouldBeEqualTo nyAktivitetskrav.uuid
+                    aktivitetskravLatestOppfolgingstilfelle.status shouldBeEqualTo AktivitetskravStatus.NY.name
+                    aktivitetskravLatestOppfolgingstilfelle.uuid shouldNotBeEqualTo nyAktivitetskrav.uuid
```

Comment:
Hvorfor får de samme uuid?

Reply:
Aktivitetskravet som finnes for tidligere oppfølgingstilfelle vil bare endre status (setter det til AUTOMATISK_OPPFYLT), så uuid er fortsatt den samme. Mens det nye aktivitetskravet vil få annen uuid (shouldNotBeEqualTo).

Reply:
Åja, jeg så ikke at den nye var `not`, da gir det mening 👍